src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -28962,6 +29256,190 @@ public void M1()
             }
         }
 
+        [Fact]
+        [WorkItem(34704, "https://github.com/dotnet/roslyn/issues/34704")]
+        public void NestedTypes_10()
+        {
+            var source1 =
+@"
+interface I1
+{
+    protected interface I2
+    {
+    }
+}
+
+class C1
+{
+    protected interface I2
+    {
+    }
+}
+
+interface I3 : I1, I1.I2
+{
+    private void M1(I1.I2 x) { }
+}
+
+interface I4 : I1, I2
+{
+    I2 MI4();
+}
+
+interface I5 : I1.I2, I1
+{
+    private void M1(I1.I2 x) { }
+}
+
+interface I6 : I2, I1
+{
+    I2 MI6();
+}
+
+class C3 : I1, I1.I2
+{
+    private void M1(I1.I2 x) { }
+}
+
+class C4 : I1, I2
+{
+    void MC4(I2 x) { }
+}
+
+class C5 : I1.I2, I1
+{
+    private void M1(I1.I2 x) { }
+}
+
+class C6 : I2, I1
+{
+    void MC6(I2 x) { }
+}
+
+class C33 : C1, C1.I2
+{
+    protected void M1(C1.I2 x) { }
+}
+
+class C44 : C1, I2
+{
+    void M1(I2 x) { }
+}
+
+interface I7 : I8
+{
+    public interface I8 : I7
+    {
+    }
+}
+
+class C7 : I8
+{
+    public interface I8
+    {
+    }
+}
+
+interface I9 : I9.I10
+{
+    public interface I10 : I9
+    {
+    }
+}
+
+interface I11
+{
+    public interface I12 : I13
+    {
+    }
+
+    public interface I13
+    {
+    }
+
+    public interface I14 : I11.I13
+    {
+    }
+}
+
+class C11
+{
+    public interface I12 : I13
+    {
+    }
+
+    public interface I13
+    {
+    }
+
+    public interface I14 : I11.I13
+    {
+    }
+}
+";
+
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular,
+                                                 targetFramework: TargetFramework.NetStandardLatest);
+
+            compilation1.VerifyDiagnostics(
+                // (16,23): error CS0122: 'I1.I2' is inaccessible due to its protection level
+                // interface I3 : I1, I1.I2
+                Diagnostic(ErrorCode.ERR_BadAccess, "I2").WithArguments("I1.I2").WithLocation(16, 23),
+                // (21,20): error CS0246: The type or namespace name 'I2' could not be found (are you missing a using directive or an assembly reference?)
+                // interface I4 : I1, I2
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "I2").WithArguments("I2").WithLocation(21, 20),
+                // (23,5): error CS0246: The type or namespace name 'I2' could not be found (are you missing a using directive or an assembly reference?)
+                //     I2 MI4();
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "I2").WithArguments("I2").WithLocation(23, 5),
+                // (26,19): error CS0122: 'I1.I2' is inaccessible due to its protection level
```

Comment:
this doesn't look correct.  That name should have been inherited.

Reply:
> this doesn't look correct.  That name should have been inherited.

Could you clarify please? Shouldn't we treat the set of base interfaces empty while we are resolving bases?
Somewhat similar to the ```class C33 : C1, C1.I2``` case, which wasn't changed in this PR.

---
In reply to: [320501659](https://github.com/dotnet/roslyn/pull/38451#discussion_r320501659) [](ancestors = 320501659)

Reply:
> Could you clarify please? Shouldn't we treat the set of base interfaces empty while we are resolving bases?

Sure, but we are not resolving bases here.  We are resolving the return type of a method.  The type declaration is
``` c#
interface I4 : I1, I2
{
    I2 MI4();
}
```
In this case the base interface `I2` isn't found so it is an error type or ignored or something.  But `I1` is resolved as a base interface, so its accessible members are inherited into the scope of `I4.  One of those members is `I2`, and that is accessible because it is `protected` and used in `I4` which is a subtype.


---
In reply to: [320506172](https://github.com/dotnet/roslyn/pull/38451#discussion_r320506172) [](ancestors = 320506172,320501659)

Reply:
> In this case the base interface I2 isn't found so it is an error type or ignored or something.  But I1 is resolved as a base interface, so its accessible members are inherited into the scope of I4.  One of those members isI2, and that is accessible because it isprotectedand used inI4` which is a subtype.

CodeFlow shows the original comment on a message corresponding to this code  ```interface I5 : I1.I2, I1```. Regardless, I think I already captured the issue in #38469.

---
In reply to: [320517670](https://github.com/dotnet/roslyn/pull/38451#discussion_r320517670) [](ancestors = 320517670,320506172,320501659)